specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2017-04-01-preview/inputs.json

```diff
@@ -660,6 +676,35 @@
         }
       }
     },
+    "RawReferenceInputDataSource": {
```

Comment:
RawReferenceInputDataSource can be below BlobReferenceInputDataSourceProperties. I think after that starts the "common" definitions for data sources.